scripts/gen-packagejson.mjs

```diff
@@ -21,6 +22,7 @@ modules
       module: `./${module}/dist/effect-${module}.esm.js`,
       import: `./${module}/dist/effect-${module}.cjs.mjs`,
       default: `./${module}/dist/effect-${module}.cjs.js`,
+      types: `./dist/declarations/src/${module}.d.ts`,
```

Comment:
all `exports` keys are order-sensitive so this should be the first one here

Reply:
Hm, in which situations would this matter? Because it already works with this in my repro.

Reply:
I'd expect that this relies on the "fallback condition" thing or however it's called but this doesn't list it as a problem: https://arethetypeswrong.github.io/?p=effect%402.0.0-next.39 . Note that this "fallback" thing is actually a bug in TS - for the time being it continues to work with it but it shouldn't. 

The issue here might be a different one. it's hard to tell on the spot without analyzing further - it's not worth spending time on it though ;p Rules-wise - it should be on the top (or at least it's usually advised for it to be at the top). If it works right now... maybe it is not worth changing it. 🤷‍♂️  Personally I'd probably change it though.